src/OMPLPlanner.cpp

```diff
@@ -128,21 +126,36 @@ bool OMPLPlanner::InitPlan(OpenRAVE::RobotBasePtr robot,
         m_simple_setup->setStartState(q_start);
 
         RAVELOG_DEBUG("Setting goal configuration.\n");
-        if (m_parameters->vgoalconfig.size() != num_dof) {
-            RAVELOG_ERROR("End configuration has incorrect DOF;"
-                          "  expected %d, got %d.\n",
-                          num_dof, m_parameters->vgoalconfig.size());
-            return false;
-        } else if (IsInOrCollision(m_parameters->vgoalconfig)) {
-            RAVELOG_ERROR("Goal configuration is in collision.\n");
-            return false;
-        }
-
-        ScopedState q_goal(m_state_space);
-        for (size_t i = 0; i < num_dof; i++) {
-            q_goal->values[i] = m_parameters->vgoalconfig[i];
-        }
-        m_simple_setup->setGoalState(q_goal);
+		std::vector<TSRChain::Ptr> goal_chains;
+		BOOST_FOREACH(TSRChain::Ptr tsr_chain, m_parameters->m_tsrchains){
+			if(tsr_chain->sampleGoal()){
+                tsr_chain->setEnv(robot->GetEnv()); // required to enable distance to TSR chains
+				goal_chains.push_back(tsr_chain);
+			}
+		}
+
+		if(goal_chains.size() > 0) {
+			TSRGoal::Ptr goaltsr = boost::make_shared<TSRGoal>(m_simple_setup->getSpaceInformation(),
+															   goal_chains,
+															   robot);
+			m_simple_setup->setGoal(goaltsr);
+		}else{
+			if (m_parameters->vgoalconfig.size() != num_dof) {
+				RAVELOG_ERROR("End configuration has incorrect DOF;"
+							  "  expected %d, got %d.\n",
+							  num_dof, m_parameters->vgoalconfig.size());
+				return false;
+			} else if (IsInOrCollision(m_parameters->vgoalconfig, dof_indices)) {
+				RAVELOG_ERROR("Goal configuration is in collision.\n");
+				return false;
+			}
+
+			ScopedState q_goal(m_state_space);
+			for (size_t i = 0; i < num_dof; i++) {
+				q_goal->values[i] = m_parameters->vgoalconfig[i];
+			}
+			m_simple_setup->setGoalState(q_goal);
+		}
```

Comment:
Fixed. Now I throw an exception if there are goal_chains and a goal config supplied.